### ISSUE TITLE
How to update serviceMonitorSelectorNilUsesHelm values when using OpenShift Prometheus Operators is used

### ISSUE THREAD

**Issue by chen-keinan** [aquasecurity/trivy-operator#1330]
### Discussed in https://github.com/aquasecurity/trivy-operator/discussions/1329

<div type='discussions-op-text'>

<sup>Originally posted by **ajay15283** June 27, 2023</sup>
If prometheus is not installed from kube-prometheus-stack and instead openshift promethus operators were used in that case how to update serviceMonitorSelectorNilUsesHelmValues to false as the metrics are not visible in grafana dashboard 

[
](https://aquasecurity.github.io/trivy-operator/v0.11.0/tutorials/grafana-dashboard/)

**From Documentation** 
Our Prometheus installation needs to be slightly customised to discover ServiceMonitors by default. Create a values.yaml file with the following configuration:

prometheus:
  prometheusSpec:
    serviceMonitorSelectorNilUsesHelmValues: true
    serviceMonitorSelector: {}
    serviceMonitorNamespaceSelector: {}

</div>

**Comment by chen-keinan** [comment]
@ajay15283 the requirements sound reasonable , feel free to pick it up

**Comment by geoffrey1330** [comment]
Hi @chen-keinan Could you please assign this issue to me.

**Comment by geoffrey1330** [comment]
@ajay15283 if am to get you right. you want the installation process for prometheus in the docs to highlight the configurations when using either kube-prometheus-stack or openshift promethus operators?

**Comment by ajay15283** [comment]
@geoffrey1330 , yes the document is clear if the prometheus is installed from kube-prometheus-stack and have no specific information for prometheus installed from other options especially if its installed as an operator

**Comment by geoffrey1330** [comment]
Alright

**Comment by geoffrey1330** [comment]
Hi @ajay15283 it seems like the process of installing prometheus via prometheus-operator is no longer supported
as stated [here](https://github.com/prometheus-operator/prometheus-operator/tree/main/helm)
they suggested everyone should make use of the [prometheus-community/kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) as it has been moved there

**Comment by geoffrey1330** [comment]
could you please move to the supported approach and close the issue.

**Comment by Prakashckesavan** [comment]
Did we got some solution to add serviceMonitorSelectorNilUsesHelmValues in openshift prometheus operator
